### DISCUSSION TITLE
Build Issue - Windows

### DISCUSSION THREAD

**Discussion by FuzzyBunnys** [ImageMagick/ImageMagick#4598] [Help]
Hello,
I'm attempting to build ImageMagick on my Windows 10 laptop so I can use the -fft command. After going through the configuration step and generating a project for visual studio, I go ahead and attempt to run the build command. It works for a bit and then encounters the following fatal errors. 

`22>LINK : fatal error LNK1181: cannot open input file 'C:\Users\NicolasKrause\Documents\ImageMagick-Windows-7\VisualMagick\lib\CORE_DB_MagickCore_.lib'
24>LINK : fatal error LNK1181: cannot open input file 'C:\Users\NicolasKrause\Documents\ImageMagick-Windows-7\VisualMagick\lib\CORE_DB_MagickCore_.lib'
23>LINK : fatal error LNK1181: cannot open input file 'C:\Users\NicolasKrause\Documents\ImageMagick-Windows-7\VisualMagick\lib\CORE_DB_MagickCore_.lib'
24>Done building project "FUZZ_encoder_label_fuzzer_StaticMTD.vcxproj" -- FAILED.
26>LINK : fatal error LNK1181: cannot open input file 'C:\Users\NicolasKrause\Documents\ImageMagick-Windows-7\VisualMagick\lib\CORE_DB_MagickCore_.lib'
25>LINK : fatal error LNK1181: cannot open input file 'C:\Users\NicolasKrause\Documents\ImageMagick-Windows-7\VisualMagick\lib\CORE_DB_MagickCore_.lib'
22>Done building project "FUZZ_encoder_plasma_fuzzer_StaticMTD.vcxproj" -- FAILED.
23>Done building project "FUZZ_encoder_pattern_fuzzer_StaticMTD.vcxproj" -- FAILED.
25>Done building project "FUZZ_encoder_gradient_fuzzer_StaticMTD.vcxproj" -- FAILED.
26>Done building project "FUZZ_encoder_fuzzer_StaticMTD.vcxproj" -- FAILED.
13>Done building project "UTIL_magick_StaticMTD.vcxproj" -- FAILED.
18>C:\Program Files\Microsoft Visual Studio\2022\Professional\MSBuild\Microsoft\VC\v170\Microsoft.CppCommon.targets(1087,5): error MSB6006: "link.exe" exited with code 1181.
18>LINK : fatal error LNK1181: cannot open input file 'C:\Users\NicolasKrause\Documents\ImageMagick-Windows-7\VisualMagick\lib\CORE_DB_MagickCore_.lib'
18>Done building project "FUZZ_huffman_decode_fuzzer_StaticMTD.vcxproj" -- FAILED.
27>LINK : fatal error LNK1181: cannot open input file 'C:\Users\NicolasKrause\Documents\ImageMagick-Windows-7\VisualMagick\lib\CORE_DB_MagickCore_.lib'
27>Done building project "FUZZ_crop_fuzzer_StaticMTD.vcxproj" -- FAILED.`

In particular it appears unable to find the CORE_DB_MagickCore_.lib, the file is not present in the specified folder, but all I did was setup the standard git clone as the instructions on the ImageMagick website say you should, so I'm not too sure where I've gone wrong.

**Comment by snibgo** [comment]
> ... as the instructions on the ImageMagick website say you should ...

What page? Please always paste the URL of any pages you cite.

I know nothing about Visual Studio, but you seem to be building a debug version. From comments I have seen on these forums, it might work better if you didn't choose a debug configuration.

**Comment by FuzzyBunnys** [comment]
I did not realize I was building a debug version! I used the following pages to attempt to build ImageMagick.

- [Windows Install](https://imagemagick.org/script/install-source.php#windows)
- [Advanced Windows Install](https://imagemagick.org/script/advanced-windows-installation.php)

I selected the multithreaded build as the first page recommends.

**Reply by richard4339** [reply]
Looks like you need to change the configuration from Debug to Release. The solution builds for me once I make that change.

**Comment by snibgo** [comment]
I think "_DB_" in your filenames indicate a debug configuration. But I could be wrong. I have no experience with Visual Studio (I use Cygwin tools).

**Reply by FuzzyBunnys** [reply]
Hi Snibgo,
I've given up on visual studio and have been following your guide re:Cygwin. I'm making progress but I'm currently stuck at the make step.
`MagickCore/nt-feature.c: In function ‘NTIsMagickConflict’:
MagickCore/nt-feature.c:96:11: warning: implicit declaration of function ‘GetLogicalDrives’ [-Wimplicit-function-declaration]
   96 |   status=(GetLogicalDrives() & (1 << ((LocaleUppercase((int)
      |           ^~~~~~~~~~~~~~~~
MagickCore/nt-feature.c: In function ‘NTAcquireTypeCache’:
MagickCore/nt-feature.c:126:3: error: unknown type name ‘HKEY’
  126 |   HKEY
      |   ^~~~
MagickCore/nt-feature.c:127:16: error: ‘HKEY’ undeclared (first use in this function)
  127 |     reg_key = (HKEY) INVALID_HANDLE_VALUE;
      |                ^~~~
MagickCore/nt-feature.c:127:16: note: each undeclared identifier is reported only once for each function it appears in
MagickCore/nt-feature.c:127:22: error: expected ‘,’ or ‘;’ before ‘INVALID_HANDLE_VALUE’
  127 |     reg_key = (HKEY) INVALID_HANDLE_VALUE;
      |                      ^~~~~~~~~~~~~~~~~~~~
MagickCore/nt-feature.c:129:3: error: unknown type name ‘LONG’
  129 |   LONG
      |   ^~~~
MagickCore/nt-feature.c:140:3: error: unknown type name ‘DWORD’
  140 |   DWORD
      |   ^~~~~
MagickCore/nt-feature.c:151:9: warning: implicit declaration of function ‘RegOpenKeyExA’ [-Wimplicit-function-declaration]
  151 |   res = RegOpenKeyExA (HKEY_LOCAL_MACHINE,
      |         ^~~~~~~~~~~~~
MagickCore/nt-feature.c:151:24: error: ‘HKEY_LOCAL_MACHINE’ undeclared (first use in this function)
  151 |   res = RegOpenKeyExA (HKEY_LOCAL_MACHINE,
      |                        ^~~~~~~~~~~~~~~~~~
MagickCore/nt-feature.c:152:59: error: ‘KEY_READ’ undeclared (first use in this function)
  152 |     "SOFTWARE\\Microsoft\\Windows NT\\CurrentVersion", 0, KEY_READ, &reg_key);
      |                                                           ^~~~~~~~
MagickCore/nt-feature.c:153:14: error: ‘ERROR_SUCCESS’ undeclared (first use in this function); did you mean ‘EXIT_SUCCESS’?
  153 |   if (res == ERROR_SUCCESS) {
      |              ^~~~~~~~~~~~~
      |              EXIT_SUCCESS
MagickCore/nt-feature.c:155:11: warning: implicit declaration of function ‘RegQueryValueExA’ [-Wimplicit-function-declaration]
  155 |     res = RegQueryValueExA(reg_key,"SystemRoot",NULL, &type,
      |           ^~~~~~~~~~~~~~~~
MagickCore/nt-feature.c:156:8: error: ‘BYTE’ undeclared (first use in this function)
  156 |       (BYTE*) system_root, &system_root_length);
      |        ^~~~
MagickCore/nt-feature.c:156:13: error: expected expression before ‘)’ token
  156 |       (BYTE*) system_root, &system_root_length);
      |             ^
MagickCore/nt-feature.c:164:15: error: expected expression before ‘)’ token
  164 |         (BYTE*)system_root, &system_root_length);
      |               ^
MagickCore/nt-feature.c:189:5: error: unknown type name ‘DWORD’
  189 |     DWORD
      |     ^~~~~
MagickCore/nt-feature.c:200:19: error: ‘ERROR_NO_MORE_ITEMS’ undeclared (first use in this function)
  200 |     while (res != ERROR_NO_MORE_ITEMS)
      |                   ^~~~~~~~~~~~~~~~~~~
MagickCore/nt-feature.c:210:15: warning: implicit declaration of function ‘RegEnumValueA’ [-Wimplicit-function-declaration]
  210 |         res = RegEnumValueA ( reg_key, registry_index, value_name,
      |               ^~~~~~~~~~~~~
MagickCore/nt-feature.c:211:47: error: expected expression before ‘)’ token
  211 |           &value_name_length, 0, &type, (BYTE*)value_data, &value_data_size);
      |                                               ^
MagickCore/nt-feature.c:191:7: warning: variable ‘value_data_size’ set but not used [-Wunused-but-set-variable]
  191 |       value_data_size,
      |       ^~~~~~~~~~~~~~~
MagickCore/nt-feature.c:386:3: warning: implicit declaration of function ‘RegCloseKey’ [-Wimplicit-function-declaration]
  386 |   RegCloseKey ( reg_key );
      |   ^~~~~~~~~~~
MagickCore/nt-feature.c:142:5: warning: variable ‘system_root_length’ set but not used [-Wunused-but-set-variable]
  142 |     system_root_length;
      |     ^~~~~~~~~~~~~~~~~~
make[1]: *** [Makefile:9278: MagickCore/libMagickCore_7_Q32HDRI_la-nt-feature.lo] Error 1
make[1]: Leaving directory '/home/NicolasKrause/ImageMagick-7.1.0-20'
make: *** [Makefile:6164: all] Error 2
`
I receive the following series of errors and I'm not too sure where to go. I've confirmed that the Cygwin paths are added to the system path as you recommend. Any assistance would be greatly appreciated!

**Comment by snibgo** [comment]
That seems to be the same as a problem I had.

See my "issues" report at https://github.com/ImageMagick/ImageMagick/issues/4638 , and my workaround.
